packages/mask/popups/pages/Wallet/Interaction/index.tsx

```diff
@@ -306,28 +306,35 @@ const Interaction = memo((props: InteractionProps) => {
 
                 if (!signRequest.includes(currentRequest.request.arguments.method)) {
                     params = compact(
-                        params.map((x) => {
-                            if (x === 'latest') {
+                        params.map((param) => {
+                            if (param === 'latest') {
                                 if (chainId === ChainId.Celo) return
-                                return x
+                                return param
                             }
 
                             return {
-                                ...x,
+                                ...param,
                                 ...(gasConfig ?
                                     mapValues(omit(gasConfig, 'gasOptionType'), (value, key) => {
                                         if (key === 'gasCurrency' || !value) return
                                         return web3_utils.toHex(value)
                                     })
                                 :   {}),
-                                gas: web3_utils.toHex(new BigNumber(gasConfig?.gas ?? x.gas).toString()),
-                                chainId: web3_utils.toHex(x.chainId),
-                                nonce: web3_utils.toHex(x.nonce),
+                                gasLimit: web3_utils.toHex(new BigNumber(gasConfig?.gas ?? param.gas).toString()),
```

Comment:
Changing field names may cause rpc calls to fail.

Reply:
the old name `gas` does not work at all. only `gasLimit` works.